rotterdam-nlds-parent-wicket/rotterdam-nlds-wicket/src/main/java/nl/rotterdam/nl_design_system/wicket/components/dialog/RdDialogBorder.java

```diff
@@ -0,0 +1,233 @@
+ * <a href="https://nldesignsystem.nl/modal-dialog/">modal dialog</a>). It must be used with a
+ * <code>&lt;dialog&gt;</code> element.
+ * </p>
+ * <p>
+ * <strong>WARNING</strong>: do not use <code>setVisible(...)</code> to show or hide the dialog. Use
+ * {@link #show(AjaxRequestTarget)}, {@link #showModal(AjaxRequestTarget)} and {@link #close(AjaxRequestTarget)}
+ * instead. These methods will use the correct JavaScript functions to manage the visibility of the dialog.
+ * </p>
+ * <p>
+ * This implementation is based on the
+ * <a href="https://designsystem.amsterdam/?path=/docs/components-containers-dialog--docs">implementation of Amsterdam</a>.
+ * <code>ModalDialog</code> of Wicket Extensions is not used as it uses the wrong markup.
+ * </p>
+ *
+ * @see RdDialogHeadingLevel
+ */
+@NlComponentState(wicketState = UNSTABLE, estafetteState = COMMUNITY, htmlCssImplementedBy = AMSTERDAM)
+public abstract class RdDialogBorder extends Border {
+
+    private static final String FOOTER_CONTENT_ID = "footerContent";
+
+    @NonNull
+    private final IModel<String> closeButtonLabelModel;
+
+    @NonNull
+    private final WebMarkupContainer content;
+
+    @NonNull
+    private final Component heading;
+
+    @NonNull
+    private final Component closeButton;
+
+    /**
+     * <p>
+     * Create a new dialog border with the given Wicket ID, title model and heading level.
+     * </p>
+     *
+     * @param id the wicket ID.
+     * @param titleModel the model with the title to shown in the dialog header.
+     * @param headingLevel the level of the heading in the dialog header. This affects the title heading element and the
+     *                     icon button.
+     */
+    public RdDialogBorder(@NonNull String id, @NonNull IModel<?> titleModel, @NonNull RdDialogHeadingLevel headingLevel) {
+        super(id, titleModel);
+        Objects.requireNonNull(titleModel);
+
+        closeButtonLabelModel = new StringResourceModel("Close", this);
+
+        setOutputMarkupId(true);
+
+        content = newContent();
+        heading = newHeading(headingLevel);
+        closeButton = newCloseButton(headingLevel);
+    }
+
+    /**
+     * <p>
+     * Create a new dialog border with the given Wicket ID, title model, close button label model and heading level.
+     * </p>
+     *
+     * @param id the wicket ID.
+     * @param titleModel the model with the title to shown in the dialog header.
+     * @param headingLevel the level of the heading in the dialog header. This affects the title heading element and the
+     *                     icon button.
+     */
+    public RdDialogBorder(@NonNull String id, @NonNull IModel<?> titleModel, @NonNull IModel<String> closeButtonLabelModel, @NonNull RdDialogHeadingLevel headingLevel) {
+        super(id, titleModel);
+        Objects.requireNonNull(titleModel);
+        Objects.requireNonNull(closeButtonLabelModel);
+
+        this.closeButtonLabelModel = closeButtonLabelModel;
```

Comment:
Inline requireNonNull checks. Preferably also with static import

```suggestion
        super(id, Objects.requireNonNull(titleModel));

        this.closeButtonLabelModel = Objects.requireNonNull(closeButtonLabelModel);
```

Reply:
Maak hier dan maar een story voor, want er zijn meer plekken waar dit niet goed gaat.

Reply:
Opgelost.